### DISCUSSION TITLE
Any further thoughts on deduped mutations with React 18?

### DISCUSSION THREAD

**Discussion by gitdagray** [TanStack/query#3802] [General] [answered]
I found this previous discussion: 

I don't think it does. mutations have side-effects on the server, they are not idempotent, so we cannot do as many optimization as with "pure" GET requests.

_Originally posted by @TkDodo in https://github.com/TanStack/query/discussions/608#discussioncomment-838630_

Use case: a request sending a refresh token stored in a secure httpOnly cookie - where the server employs re-use detection - should only be sent once, and the request returns new auth state. I have a PersistLogin wrapper component that sends this request when a user wants to persist their login and has lost their state due to refreshing or navigating away and returning. Before React 18, useEffect triggered this request once with no issues. I'm now applying a useRef fix to avoid the double call in strict mode. Maybe I need to stick with useRef but I'm wondering if it would be possible for React Query to look further into mutation deduplication.

**Comment by TkDodo** [comment, marked as answer]
I still think this is out of scope for react-query; the issue seems to be react firing effects twice in strict mode (only for dev builds), so the workarounds are either to ignore it (because it has not production impact) or to work around it with the ref.

**Reply by gitdagray** [reply]
Thank you for your reply 🙏
